M2/Macaulay2/m2/exports.m2

```diff
@@ -630,7 +631,6 @@ export {
 	"difference",
 	"dim",
 	"directSum",
-	"disassemble",
```

Comment:
I think you should keep this exported, especially since it's an easier way to traverse Pseudocode.

Also, is there a reason why `disassemble` returns mostly sequences, but one list near the beginning, and doesn't disassemble all the way? See:
```m2

i2 : X = disassemble pseudocode(() -> try true);

i3 : class \ X

o3 = (String, List)

o3 : Sequence

i4 : class \ X#1#4

o4 = (String, Pseudocode)

o4 : Sequence

i5 : disassemble X#1#4
stdio:5:11:(3): error: expected pseudocode

i6 : disassemble X#1#4#1

o6 = (try, {global-fetch 219, (then, null), (else, null)})

o6 : Sequence
```

Reply:
it doesn't disassemble all the way because that's how it works. `Pseudocode` is not a list, so if you want to see one of its parts, you have to disassemble only at one level.
I don't think `disassemble` should be exported, it's only meant to be used internally.
`disassemble` returns a sequence which is simply a list of objects to be printed next to each other.

Reply:
What is the rationale behind these decisions? To me they seem arbitrary. `disassemble` used to print the S-expression (aka "lisp version") of the code for the user, and your proposed `Pseudocode` type is pretty but doesn't replace that.

Reply:
`toString` will produce the same lisp version as what `disassemble` used to do, with perhaps a few useless parentheses removed. The rest will be a string just like `disassemble` used to produce, which means fairly useless per se (why should one have to parse a string?). I believe the current format is much more structured. I'm sure one can it be improved, but the principle seems sound to me.

Reply:
This is _not lisp_:
```m2
i3 : toString pseudocode(() -> (4 + 4) / 2)

o3 = function (restargs false numparms 0 framesize 0 frameID 11325 body 2-OP / (2-OP + (4 4) 2))
```
This is:
```m2
i1 : disassemble pseudocode(() -> (4 + 4) / 2)

o1 = (function restargs: false numparms: 0 framesize: 0 frameID: 8748 (2-OP / (2-OP + 4 4) 2))
```
One pair of parentheses are replaced by "body", but you've also moved every other parenthesis. You seem to prefer the `Pseudocode` display, so why do you insist on changing the lisp format?

I agreed with you about removing the colons after "then" and "else" but you also removed the colons after "restargs", etc. which definitely make this output look _less_ structured.

Reply:
the problem with colons is, one can put them everywhere or remove them everywhere but anything in between is fairly arbitrary.